test/Datadog.Trace.Tests/TracerTests.cs

```diff
@@ -337,5 +340,52 @@ public void SetServiceName(string envServiceName, string tracerServiceName, stri
             // reset the environment variable to its original values (if any) when done
             Environment.SetEnvironmentVariable(name, originalEnv);
         }
+
+        [Fact]
+        public void OriginHeader_RootSpanTag()
+        {
+            const ulong traceId = 9;
+            const ulong spanId = 7;
+            const SamplingPriority samplingPriority = SamplingPriority.UserKeep;
+            const string origin = "synthetics";
+
+            var propagatedContext = new SpanContext(traceId, spanId, samplingPriority, origin: origin);
+            Assert.Equal(origin, propagatedContext.Origin);
+
+            using var firstSpan = _tracer.StartActive("First Span", propagatedContext);
+            Assert.True(firstSpan.Span.IsRootSpan);
+            Assert.Equal(origin, firstSpan.Span.Context.Origin);
+            Assert.Equal(origin, firstSpan.Span.GetTag(Tags.Origin));
+
+            using var secondSpan = _tracer.StartActive("Child", firstSpan.Span.Context);
+            Assert.False(secondSpan.Span.IsRootSpan);
+            Assert.Equal(origin, secondSpan.Span.Context.Origin);
+            Assert.Null(secondSpan.Span.GetTag(Tags.Origin));
+        }
+
+        [Fact]
+        public void OriginHeader_InjectFromChildSpan()
+        {
+            const ulong traceId = 9;
+            const ulong spanId = 7;
+            const SamplingPriority samplingPriority = SamplingPriority.UserKeep;
+            const string origin = "synthetics";
+
+            var propagatedContext = new SpanContext(traceId, spanId, samplingPriority, origin: origin);
+
+            using var firstSpan = _tracer.StartActive("First Span", propagatedContext);
+            using var secondSpan = _tracer.StartActive("Child", firstSpan.Span.Context);
+
+            IHeadersCollection headers = WebRequest.CreateHttp("http://localhost").Headers.Wrap();
+
+            SpanContextPropagator.Instance.Inject(firstSpan.Span.Context, headers);
```

Comment:
I think you meant to inject from the child span here:

```suggestion
            SpanContextPropagator.Instance.Inject(secondSpan.Span.Context, headers);
```

Reply:
Yeah, thank you for the catch there...